src/app/lists/edit_modal.tsx

```diff
@@ -0,0 +1,155 @@
+        generateDefaultData()
+    );
+
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    const headersAndTooltips: [[string, string], [string, string]][] = [];
+    const headersThatHaveTooltips = headers.filter(([key]) => key !== "item_name");
+
+    for (let index = 0; index < headersThatHaveTooltips.length; index++) {
+        headersAndTooltips.push([headersThatHaveTooltips[index], tooltips[index]]);
+    }
+
+    return (
+        <Modal header="Edit List" headerId="editList" isOpen={data !== undefined} onClose={onClose}>
+            <ModalInner>
+                <h3>Description</h3>
+                <TextInput
+                    defaultValue={toSubmit.item_name ?? ""}
+                    onChange={(event) => {
+                        const text = event.target.value as string;
+                        setToSubmit({ ...toSubmit, item_name: text });
+                    }}
+                    helperText="Description"
+                />
+                {headersAndTooltips.map(([[key, label], [tooltipKey]]) => {
+                    return (
+                        <DisplayContents key={key}>
+                            <h3>{label}</h3>
+                            <DataWithTooltipDiv>
+                                <TextInput
+                                    defaultValue={toSubmit[key] ?? ""}
+                                    helperText="Quantity"
+                                    onChange={(event) => {
+                                        const text = event.target.value as string;
+                                        setToSubmit({ ...toSubmit, [key]: text });
+                                    }}
+                                />
+                                <TextInput
+                                    defaultValue={toSubmit[tooltipKey] ?? ""}
+                                    helperText="Notes"
+                                    onChange={(event) => {
+                                        const text = event.target.value as string;
+                                        setToSubmit({ ...toSubmit, [tooltipKey]: text });
+                                    }}
+                                />
+                            </DataWithTooltipDiv>
+                        </DisplayContents>
+                    );
+                })}
+                <Button
+                    variant="contained"
+                    color="primary"
+                    onClick={async () => {
+                        let operation;
+
+                        if (data === null) {
+                            operation = supabase.from("lists").insert(toSubmit);
+                        } else {
+                            operation = supabase
+                                .from("lists")
+                                .update(toSubmit)
+                                .eq("primary_key", toSubmit.primary_key)
+                                .select();
```

Comment:
When I want to insert a new row, I got Failed to load resource: the server responded with a status of 409. Is that just my laptop?

Reply:
Also it seems like I cant edits the rows as well. Is this still in development?

Reply:
That's an issue with the database, modifications were working previously, I'll have a look into it